File: ValidParentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        l=[]
        for i in s:
            if i == '(' or i == '{' or i == '[':
                l.append(i)
            elif (i == ')' or i == '}' or i == ']') and (len(l) != 0):
                if (i == ')' and l[-1] == '(') or (i == '}' and l[-1] == '{') or (i == ']' and l[-1] == '['):
                    l.pop()
                else:
                    return False
            else:
                return False
        return len(l) == 0
                
        

File: ValidParentheses.java
class Solution {
    public boolean isValid(String s) {
        
        Stack<Character> stack = new Stack<>();

        int n = s.length();

        for(int i = 0;i < n;i++){
            if(s.charAt(i) == '(' || s.charAt(i) == '{' || s.charAt(i) == '[')stack.push(s.charAt(i));
            else{
                if(s.charAt(i) == ')'){
                    if(!stack.empty() && stack.peek() == '(')stack.pop();
                    else return false;
                }

                if(s.charAt(i) == '}'){
                    if(!stack.empty() && stack.peek() == '{')stack.pop();
                    else return false;
                }

                if(s.charAt(i) == ']'){
                    if(!stack.empty() && stack.peek() == '[')stack.pop();
                    else return false;
                }
            }
        }

        return stack.empty();
    }
}
